hoomd/hpmc/external/wall.py

```diff
@@ -61,6 +61,26 @@ def _attach(self):
         if not isinstance(integrator, hoomd.hpmc.integrate.HPMCIntegrator):
             raise RuntimeError('Walls require a valid HPMC integrator.')
 
+        # check that shape-wall overlap checks are implemented, error if not
+        supported_shape_wall_overlap_checks = {
```

Comment:
This will check the that the wall/shape combo is supported at initialization time. If possible, also check on setting new walls to error in cases where the user starts with a supported combo and then assigns an unsupported one. This could occur for example when starting with an empty wall list and a `ConvexSpheropolyhedron` integrator, then adding a cylinder wall.

Reply:
Given the nature of `_WallsMetaList` we need to change the available types when attaching and remove unavailable types at attaching. Currently this is not a feature from the class so it would need to be monkey patching in a sense here which would involve removing and modifying some attributes of the list.

Reply:
Good catch. I refactored this check into a separate class method that is used here and in the walls setter.